ddtrace/contrib/unittest/patch.py

```diff
@@ -103,85 +279,327 @@ def patch():
+    span.set_tag_str(SPAN_KIND, KIND)
+
+    span.set_tag_str(test.COMMAND, test_suite_span.get_tag(test.COMMAND))
+    span.set_tag_str(test.FRAMEWORK, FRAMEWORK)
+    span.set_tag_str(test.FRAMEWORK_VERSION, _get_runtime_and_os_metadata()[RUNTIME_VERSION])
+    span.set_tag_str(test.TYPE, SpanTypes.TEST)
+
+    span.set_tag_str(test.NAME, test_name)
+    span.set_tag_str(test.SUITE, test_suite_name)
+    span.set_tag_str(test.MODULE, test_suite_span.get_tag(test.MODULE))
+    span.set_tag_str(test.MODULE_PATH, test_suite_span.get_tag(test.MODULE_PATH))
+    span.set_tag_str(test.STATUS, test.Status.FAIL.value)
+    span.set_tag_str(test.CLASS_HIERARCHY, test_suite_name)
+
+    _CIVisibility.set_codeowners_of(_extract_test_file_name(instance), span=span)
+
+    _store_test_span(instance, span)
+    return span
+
+
+def _finish_span(current_span):
+    current_status = current_span.get_tag(test.STATUS)
+    parent_span = current_span._parent
+    if current_status and parent_span:
+        _update_status_item(parent_span, current_status)
+    elif not current_status:
+        current_span.set_tag_str(test.SUITE, test.Status.FAIL.value)
+    current_span.finish()
+
+
+def handle_cli_run(func, instance, args, kwargs):
+    test_session_span = None
+    if _is_invoked_by_cli(instance):
+        if not hasattr(_CIVisibility, "_unittest_data"):
+            _CIVisibility._unittest_data = {"suites": {}, "modules": {}}
+        for parent_module in instance.test._tests:
+            for module in parent_module._tests:
+                _populate_suites_and_modules(
+                    module, _CIVisibility._unittest_data["suites"], _CIVisibility._unittest_data["modules"]
+                )
+
+        test_session_span = _start_test_session_span(instance)
+        _CIVisibility._datadog_entry = "cli"
+        _CIVisibility._datadog_session_span = test_session_span
+
+    try:
+        result = func(*args, **kwargs)
+    except SystemExit as e:
+        if _CIVisibility.enabled and test_session_span:
+            _finish_remaining_suites_and_modules(
+                _CIVisibility._unittest_data["suites"], _CIVisibility._unittest_data["modules"]
+            )
+            _finish_span(test_session_span)
+        raise e
+    return result
+
+
+def handle_text_test_runner_wrapper(func, instance, args, kwargs):
+    """
+    Creates session and module spans if the unittest is called through the `TextTestRunner` method
+    """
+    if _is_invoked_by_cli(instance):
+        return func(*args, **kwargs)
+    _CIVisibility._datadog_entry = "TextTestRunner"
+    try:
+        if not hasattr(_CIVisibility, "_datadog_session_span"):
+            _CIVisibility._datadog_session_span = _start_test_session_span(instance)
+            _CIVisibility._datadog_expected_sessions = 0
+            _CIVisibility._datadog_finished_sessions = 0
+        _CIVisibility._datadog_expected_sessions += 1
+        result = func(*args, **kwargs)
+    except Exception as e:
```

Comment:
Can this exception handler expect a more specific exception type? If so, making that change would reduce the possibility of errors being unintentionally swallowed.